sci-libs/cantera/cantera-2.4.0.ebuild

```diff
@@ -0,0 +1,130 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python3_{4,5,6} )
+
+inherit desktop python-single-r1 scons-utils toolchain-funcs
+
+DESCRIPTION="Object-oriented tool suite for chemical kinetics, thermodynamics, and transport"
+HOMEPAGE="http://www.cantera.org"
+SRC_URI="https://github.com/Cantera/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+cti cxx fortran pch +python static-libs test"
+
+REQUIRED_USE="
+	|| ( cxx python )
+	cti? ( ${PYTHON_REQUIRED_USE} )
+	fortran? ( cxx )
+	python? ( cti )
+	static-libs? ( cxx )
+	^^ ( $(python_gen_useflags 'python3*') )
+	"
+
+RDEPEND="
+	python? (
+		dev-python/numpy[${PYTHON_USEDEP}]
+	)
+"
+
+DEPEND="
+	${RDEPEND}
+	dev-cpp/eigen
+	dev-libs/boost
```

Comment:
No `:=` on boost?

Also is it used as header-only library? Because otherwise it needs to go to RDEP as well.

Reply:
The official package documentation in ["Software used by Cantera" section](https://cantera.org/compiling/dependencies.html) mentions that 

> * Boost
> ...
>    - Only the “header-only” portions of Boost are required. Cantera does not currently depend on any of the compiled Boost libraries.

and the [Cantera 2.4.0 changelog](https://cantera.org/blog/cantera-240-released.html) states that
> #### Changes to existing capabilities
> ...
> * Remove the requirement for users to have Boost headers installed when building against the Cantera library

So it seems that Boost headers are used now only to build Cantera library itself and aren't required for runtime of C++ module.

Reply:
I checked and like for FMT library boost is required by some of cantera package header(s) (in common case but there is a  "NO_BOOST"-like ifndefine in one of  header file). But both shouldn't require to rebuild cantera package after their SLOT update (cantera C++ header files use only boost and libfmt headers).

Sundials is required by C++ library and portage show message about preserved libs in sundials was removed:
```
!!! existing preserved libs:
...
 *  - /usr/lib64/libsundials_cvodes.so.3
 *  - /usr/lib64/libsundials_cvodes.so.3.1.0
 *      used by /usr/lib64/libcantera_shared.so.2.4.0 (sci-libs/cantera-2.4.0)
 *      used by /usr/lib64/python3.6/site-packages/cantera/_cantera.cpython-36m-x86_64-linux-gnu.so (sci-libs/cantera-2.4.0)
```
although I didn't find the presented python example that require sundials library to operate python module (to build it the C++ library is required).

Eigen is mentioned in one of cantera header.

As all of them only required by C++ library that is build initially and is used to build python module maybe it worth then to remove `cxx` USE-flag and move `sci-libs/sundials:0=`, `dev-libs/libfmt`, `dev-libs/boost`, `dev-cpp/eigen` to RDEP to simplify ebuild? Otherwise if USE="-cxx" them are not required as RDEPS.